services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py

```diff
@@ -0,0 +1,161 @@
+            sa.select([comp_tasks]).where(
+                (comp_tasks.c.project_id == str(project_id))
+                & (comp_tasks.c.node_class == NodeClass.COMPUTATIONAL)
+            )
+        ):
+            task_db = CompTaskAtDB.from_orm(row)
+            tasks.append(task_db)
+
+        return tasks
+
+    @log_decorator(logger=logger)
+    async def publish_tasks_from_project(
+        self,
+        project: ProjectAtDB,
+        director_client: DirectorV0Client,
+    ) -> None:
+        # start by removing the old tasks if they exist
+        await self.connection.execute(
+            sa.delete(comp_tasks).where(comp_tasks.c.project_id == str(project.uuid))
+        )
+        # create the tasks
+        workbench = project.workbench
+        internal_id = 1
+        for node_id in workbench:
+            node: Node = workbench[node_id]
+
+            service_key_version = ServiceKeyVersion(
+                key=node.key,
+                version=node.version,
+            )
+            node_class = to_node_class(service_key_version.key)
+            node_details: ServiceDockerData = None
+            node_extras: ServiceExtras = None
+            if node_class == NodeClass.FRONTEND:
+                node_details = _get_fake_service_details(service_key_version)
+            else:
+                node_details = await director_client.get_service_details(
+                    service_key_version
+                )
+                node_extras: ServiceExtras = await director_client.get_service_extras(
+                    service_key_version
+                )
+            requires_mpi = False
+            requires_gpu = False
+            if node_extras:
+                requires_gpu = node_extras.node_requirements == NodeRequirement.GPU
+                requires_mpi = node_extras.node_requirements == NodeRequirement.MPI
+            image = Image(
+                name=service_key_version.key,
+                tag=service_key_version.version,
+                requires_gpu=requires_gpu,
+                requires_mpi=requires_mpi,
+            )
+
+            task_db = CompTaskAtDB(
+                project_id=project.uuid,
+                node_id=node_id,
+                schema=NodeSchema(
+                    inputs=node_details.inputs, outputs=node_details.outputs
+                ),
+                inputs=node.inputs,
+                outputs=node.outputs,
+                image=image,
+                submit=datetime.utcnow(),
+                state=RunningState.PUBLISHED
+                if node_class == NodeClass.COMPUTATIONAL
+                else RunningState.NOT_STARTED,
+                internal_id=internal_id,
+                node_class=node_class,
+            )
+            internal_id = internal_id + 1
+
+            await self.connection.execute(
+                insert(comp_tasks).values(
+                    **task_db.dict(by_alias=True, exclude_unset=True)
+                )
+            )
```

Comment:
Are these comp_tasks being deleted and added again?

We had an issue with mixing Comp service -> Dynamic service -> Comp service where it was required to update the state without removing them.

Reply:
so. they are inserted when you first press play.
and actually you are right that is an issue! thanks!

Reply:
In the sidecar implementation, when the user pressed run, the comp_tasks were updated and only the status of the comp_task was unset and made ready to run again, thus conserving the incoming data from the ports.

Reply:
ok after double checking there is no risk here. on line 80-82 the tasks are first removed.

Reply:
so the issue you were talking about in the first place might happen. will have to check it out cause I think this dependency should not exist if possible